XRTK-Core/Packages/com.xrtk.core/Runtime/Providers/Controllers/Simulation/Hands/SimulatedHandDataConverter.cs

```diff
@@ -0,0 +1,292 @@
+        /// Will get clamped to [current,1] where 1 means the hand has reached the target pose.
+        /// </summary>
+        public float TargetPoseBlending
+        {
+            get => targetPoseBlending;
+            private set => targetPoseBlending = Mathf.Clamp(value, targetPoseBlending, 1.0f);
+        }
+
+        /// <summary>
+        /// The current hand data produced by the current simulation state.
+        /// </summary>
+        public HandData HandData { get; } = new HandData();
+
+        /// <summary>
+        /// Current rotation of the hand.
+        /// </summary>
+        public Vector3 HandRotateEulerAngles { get; private set; } = Vector3.zero;
+
+        /// <summary>
+        /// Current random offset to simulate tracking inaccuracy.
+        /// </summary>
+        public Vector3 JitterOffset { get; private set; } = Vector3.zero;
+
+        /// <summary>
+        /// Gets simulated hand data for a <see cref="MixedRealityHandController"/>.
+        /// </summary>
+        /// <param name="position">The simulated camera space position of the hand controller.</param>
+        /// <param name="deltaRotation">The rotation delta applied to the hand since last update.</param>
+        /// <returns>Updated simulated hand data.</returns>
+        public HandData GetSimulatedHandData(Vector3 position, Vector3 deltaRotation)
+        {
+            HandData.PointerPose = MixedRealityPose.ZeroIdentity;
+
+            // Read keyboard / mouse input to determine the root pose delta since last frame.
+            var rootPoseDelta = new MixedRealityPose(position, Quaternion.Euler(deltaRotation));
+
+            // Calculate pose changes and compute timestamp for hand tracking update.
+            var poseAnimationDelta = handPoseAnimationSpeed * Time.deltaTime;
+            var timeStamp = handUpdateStopWatch.TimeStamp;
+
+            // Update simulated hand states using collected data.
+            var newTargetPose = GetTargetHandPose();
+
+            HandleSimulationInput(rootPoseDelta);
+
+            if (!string.Equals(newTargetPose.Id, Pose.Id) && HandData.IsTracked)
+            {
+                previousPose = Pose;
+                TargetPose = newTargetPose;
+            }
+
+            TargetPoseBlending += poseAnimationDelta;
+
+            if (HandData.UpdatedAt != timeStamp)
+            {
+                HandData.UpdatedAt = timeStamp;
+                if (HandData.IsTracked)
+                {
+                    UpdatePoseFrame();
+                }
+            }
+
+            return HandData;
+        }
+
+        private void HandleSimulationInput(MixedRealityPose handRootPose)
+        {
+            // If the hands state is changing from "not tracked" to being tracked, reset its position
+            // to the current mouse position and default distance from the camera.
+            if (!HandData.IsTracked)
+            {
+                Vector3 mousePos = Input.mousePosition;
```

Comment:
I wonder if it's possible to use the input from the `MouseDataProvider` ?

Reply:
Could be possible, it would couple simulation to having an active MouseDataProvider though. Let's wait with this and see how we solve the simulation inputs in general.